server/src/internalClusterTest/java/org/elasticsearch/search/fetch/subphase/highlight/HighlighterSearchIT.java

```diff
@@ -2113,6 +2115,45 @@ public void testMultiMatchQueryHighlight() throws IOException {
         }
     }
 
+    public void testCombinedFieldsQueryHighlight() throws IOException {
+        XContentBuilder mapping = XContentFactory.jsonBuilder().startObject().startObject("_doc")
+                .startObject("properties")
+                    .startObject("field1")
+                        .field("type", "text")
+                        .field("index_options", "offsets")
+                        .field("term_vector", "with_positions_offsets")
+                    .endObject()
+                    .startObject("field2")
+                        .field("type", "text")
+                        .field("index_options", "offsets")
+                        .field("term_vector", "with_positions_offsets")
+                    .endObject()
+                .endObject()
+                .endObject().endObject();
+        assertAcked(prepareCreate("test").setMapping(mapping));
+        ensureGreen();
+
+        client().prepareIndex("test")
+            .setSource("field1", "The quick brown fox jumps over", "field2", "The quick brown fox jumps over")
+            .get();
+        refresh();
+
+        for (String highlighterType : ALL_TYPES) {
+            CombinedFieldsQueryBuilder multiMatchQueryBuilder = combinedFieldsQuery("the quick brown fox", "field1", "field2");
+            SearchSourceBuilder source = searchSource()
+                    .query(multiMatchQueryBuilder)
+                    .highlighter(highlight()
+                            .highlighterType(highlighterType)
+                            .field(new Field("field1").requireFieldMatch(true).preTags("<field1>").postTags("</field1>")));
+
+            SearchResponse searchResponse = client().search(searchRequest("test").source(source)).actionGet();
+            assertHitCount(searchResponse, 1L);
+            assertHighlight(searchResponse, 0, "field1", 0,
+                anyOf(equalTo("<field1>The quick brown fox</field1> jumps over"),
```

Comment:
I think it can never be `"<field1>The quick brown fox</field1> jumps over"`? No? Isn't this only applicable for phrase and phrase_prefix search?

Reply:
Good point, I'll remove the first option here.